code/game/objects/structures/barrel.dm

```diff
@@ -1,15 +1,52 @@
 /obj/structure/reagent_dispensers/barrel
-	name                = "barrel"
-	desc                = "A stout barrel for storing large amounts of liquids or substances."
-	icon                = 'icons/obj/structures/barrel.dmi'
-	icon_state          = ICON_STATE_WORLD
-	anchored            = TRUE
-	atom_flags          = ATOM_FLAG_CLIMBABLE | ATOM_FLAG_OPEN_CONTAINER
-	matter              = null
-	material            = /decl/material/solid/organic/wood
-	material_alteration = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_DESC
-	wrenchable          = FALSE
-	storage             = /datum/storage/hopper/industrial
+	name                      = "barrel"
+	desc                      = "A stout barrel for storing large amounts of liquids or substances."
+	icon                      = 'icons/obj/structures/barrel.dmi'
+	icon_state                = ICON_STATE_WORLD
+	anchored                  = TRUE
+	atom_flags                = ATOM_FLAG_CLIMBABLE | ATOM_FLAG_OPEN_CONTAINER
+	matter                    = null
+	material                  = /decl/material/solid/organic/wood
+	material_alteration       = MAT_FLAG_ALTERATION_COLOR | MAT_FLAG_ALTERATION_NAME | MAT_FLAG_ALTERATION_DESC
+	wrenchable                = FALSE
+	storage                   = /datum/storage/barrel
+	amount_dispensed          = 10
+	possible_transfer_amounts = @"[10,25,50,100]"
+	volume                    = 7500
+	atom_flags                = ATOM_FLAG_CLIMBABLE
+	movable_flags             = MOVABLE_FLAG_WHEELED
+
+/obj/structure/reagent_dispensers/barrel/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD
```

Comment:
```suggestion
	. = ..()
	if(mapload)
		return INITIALIZE_HINT_LATELOAD
```
see other comment, this seems to solve the unit testing issue and gives more reliable behavior. if it's created or constructed outside of mapload stuff should have to be manually inserted IMO